genesis/engine/entities/rigid_entity/rigid_entity.py

```diff
@@ -1529,6 +1529,12 @@ def get_joint(self, name=None, uid=None):
             for joint in self._joints:
                 if joint.name == name:
                     return joint
+            for joint in self._joints:
+                if f"{name}," in joint.name or f",{name}" in joint.name:
+                    gs.logger.warning(
```

Comment:
I do not think it is a good idea to just report a warning and return the merged joint silently. I would advice to raise an exception by default, and add an extra optional argument to specify whether merged joint should be considered.

Besides, the logic to detect whether it is a merged joint or not is brittle, and it is not possible to split them back into individual joints. It would advise to modify `RigidJoint` to add a new field `original_joints` or `true_joints`, that keeps track of the original joints, without using them for computation (each of them being some `VirtualJoint` instances that would only contain information can be recovered). This way, it would be possible to return a joint object that would be helpful for the end-user systematically.